mod/metrics/lib/javascript/script_metrics.js.coffee

```diff
@@ -72,15 +117,21 @@ DIGITS_AFTER_DECIMAL = 2
 
 tallyWeights = (tbody, hash) ->
   multiplier = 10**DIGITS_AFTER_DECIMAL
-  total =  0
+  aux = valuesAreValid(hash, multiplier)
+  if aux.valid 
+    total = aux.total / multiplier
+    publishWeightTotal(tbody, hash, total)
+    aux.valid = total > 99.90 and total <= 100.09
+  return aux.valid
```

Comment:
I think this can be more compact, eg:
```
aux = valuesAreValid(hash, multiplier)
return false unless aux.valid
total = aux.total / multiplier
publishWeightTotal(tbody, hash, total)
total > 99.90 and total <= 100.09
```